File: src/main/java/io/femo/http/transport/http2/Http20Exception.java
package io.femo.http.transport.http2;

import io.femo.http.Constants;
import org.jetbrains.annotations.NonNls;

/**
 * Created by felix on 9/16/16.
 */
public class Http20Exception extends RuntimeException {

    private int errorCode = Constants.Http20.ErrorCodes.PROTOCOL_ERROR;

    public Http20Exception(@NonNls String message) {
        super(message);
    }

    public Http20Exception(String message, Throwable cause) {
        super(message, cause);
    }

    public Http20Exception(Throwable cause) {
        super(cause);
    }

    public Http20Exception(@NonNls String message, int errorCode) {
        super(message);
        this.errorCode = errorCode;
    }

    public Http20Exception(String message, Throwable cause, int errorCode) {
        super(message, cause);
        this.errorCode = errorCode;
    }

    public Http20Exception(Throwable cause, int errorCode) {
        super(cause);
        this.errorCode = errorCode;
    }

    public int getErrorCode() {
        return errorCode;
    }
}

File: src/main/java/org/web25/http/transport/http2/Http20Exception.kt
package org.web25.http.transport.http2

import org.jetbrains.annotations.NonNls
import org.web25.http.Constants

/**
 * Created by felix on 9/16/16.
 */
class Http20Exception : RuntimeException {

    var errorCode = Constants.Http20.ErrorCodes.PROTOCOL_ERROR
        private set

    constructor(@NonNls message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}

    constructor(cause: Throwable) : super(cause) {}

    constructor(@NonNls message: String, errorCode: Int) : super(message) {
        this.errorCode = errorCode
    }

    constructor(message: String, cause: Throwable, errorCode: Int) : super(message, cause) {
        this.errorCode = errorCode
    }

    constructor(cause: Throwable, errorCode: Int) : super(cause) {
        this.errorCode = errorCode
    }
}
